### PR TITLE
[Refactor] add `init_model` and `inference_topdown` apis

### PR DESCRIPTION
**Pull request by ly015**
## Motivation

Add model inference apis

## Modification

- Add apis: `init_model`, `inference_topdown`
- Add `demo/image_demo.py`
- Change default bbox format from 'xywh' to 'xyxy' to align with other codebases
- Modify `PackPoseInput` to pack training-related data (e.g. heatmaps, labels and keypoint_weights) as torch.Tensor and other data (e.g. keypoints, bboxes) as numpy.ndarray
- Add a demo config for hrnet+coco at `configs/tests/hrnet_w32_coco_256x192.py`

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpose/apis/inference.py [hunk 1]
```diff
@@ -1,848 +1,177 @@
...
+from mmpose.datasets.datasets.utils import parse_pose_metainfo
+from mmpose.models.builder import build_pose_estimator
 
-os.environ['KMP_DUPLICATE_LIB_OK'] = 'TRUE'
 
-
-def init_pose_model(config, checkpoint=None, device='cuda:0'):
-    """Initialize a pose model from config file.
+def dataset_meta_from_config(config: Config,
+                             dataset: str = 'train') -> Optional[dict]:
```
Comment (liqikai9) [inline mmpose/apis/inference.py:20]: How about using `dataset_mode` instead of `dataset`?

File: mmpose/apis/inference.py [hunk 2]
```diff
@@ -1,848 +1,177 @@
...
+        config (str, :obj:`Path`, or :obj:`mmengine.Config`): Config file path,
+            :obj:`Path`, or the config object.
+        dataset (str): Specify the dataset of which to get the metainfo.
+            Options are ``'train'``, ``'val'`` and ``'test'``. Defaults to
+            ``'train'``
 
     Returns:
-        nn.Module: The constructed detector.
+        dict, optional: The dataset metainfo. See
+        ``mmpose.datasets.datasets.parse_pose_metainfo`` for more details.
```
Comment (liqikai9) [inline mmpose/apis/inference.py:32]: How about `mmpose.datasets.datasets.parse_pose_metainfo` -> `mmpose.datasets.datasets.utils.parse_pose_metainfo`?

File: mmpose/apis/inference.py [hunk 3]
```diff
@@ -1,848 +1,177 @@
...
-        sample = imgs_or_paths[0] if use_multi_frames else imgs_or_paths
-        if isinstance(sample, str):
-            width, height = Image.open(sample).size
+        if 'dataset_meta' in ckpt.get('meta', {}):
+            # checkpoint from mmpose 1.x
+            model.dataset_meta = ckpt['meta']['dataset_meta']
         else:
-            height, width = sample.shape[:2]
-        person_results = [{'bbox': np.array([0, 0, width, height])}]
+            dataset_meta = dataset_meta_from_config(config, dataset='test')
```
Comment (liqikai9) [inline mmpose/apis/inference.py:104]: `dataset='test'` -> `dataset_mode='test'`
